webapp/components/ContentMenu/ContentMenu.spec.js

```diff
@@ -0,0 +1,463 @@
+        expect(
+          wrapper
+            .findAll('.ds-menu-item')
+            .filter(item => item.text() === 'post.menu.edit')
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8')
+      })
+
+      it('can delete the contribution', () => {
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'post.menu.delete')
+          .at(0)
+          .trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
+      })
+    })
+
+    describe('admin can', () => {
+      it('pin unpinned post', () => {
+        getters['auth/isAdmin'] = () => true
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+            pinnedBy: null,
+          },
+        })
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'post.menu.pin')
+          .at(0)
+          .trigger('click')
+        expect(wrapper.emitted('pinPost')).toEqual([
+          [
+            {
+              id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+              pinnedBy: null,
+            },
+          ],
+        ])
+      })
+
+      it('unpin pinned post', () => {
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+            pinnedBy: 'someone',
+          },
+        })
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'post.menu.unpin')
+          .at(0)
+          .trigger('click')
+        expect(wrapper.emitted('unpinPost')).toEqual([
+          [
+            {
+              id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+              pinnedBy: 'someone',
+            },
+          ],
+        ])
+      })
+    })
+
+    describe('owner of comment can', () => {
+      let wrapper
+      beforeEach(() => {
+        wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'comment',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+      })
+      it('edit the comment', () => {
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'comment.menu.edit')
+          .at(0)
+          .trigger('click')
+        expect(wrapper.emitted('showEditCommentMenu')).toEqual([[true]])
+      })
+      it('delete the comment', () => {
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'comment.menu.delete')
+          .at(0)
+          .trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-11T21:58:57Z" title="Monday, November 11th 2019, 10:58:57 pm +01:00">Nov 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,543 @@
+import { config, mount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import VTooltip from 'v-tooltip'
+import Styleguide from '@human-connection/styleguide'
+import ContentMenu from './ContentMenu.vue'
+
+const localVue = createLocalVue()
+
+localVue.use(Styleguide)
+localVue.use(VTooltip)
+localVue.use(Vuex)
+
+config.stubs['router-link'] = '<span><slot /></span>'
+
+let getters, mutations, mocks, menuToggle, openModalSpy
+
+describe('ContentMenu.vue', () => {
+  beforeEach(() => {
+    mocks = {
+      $t: jest.fn(str => str),
+      $i18n: {
+        locale: () => 'en',
+      },
+      $router: {
+        resolve: jest.fn(obj => {
+          obj.href = '/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8'
+          return obj
+        }),
+        push: jest.fn(),
+      },
+    }
+  })
+
+  describe('mount', () => {
+    mutations = {
+      'modal/SET_OPEN': () => jest.fn(),
+    }
+    getters = {
+      'auth/isModerator': () => false,
+      'auth/isAdmin': () => false,
+    }
+
+    const openContentMenu = (values = {}) => {
+      const store = new Vuex.Store({ mutations, getters })
+      const wrapper = mount(ContentMenu, {
+        propsData: {
+          ...values,
+        },
+        mocks,
+        store,
+        localVue,
+      })
+      menuToggle = wrapper.find('.content-menu-trigger')
+      menuToggle.trigger('click')
+      return wrapper
+    }
+
+    describe('contribution', () => {
+      it('owner can edit and delete contribution', () => {
+        const wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        let items = wrapper.findAll('.ds-menu-item')
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+        expect(items).toHaveLength(2)
+        expect(items.at(0).text()).toBe('post.menu.edit')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8')
+        expect(items.at(1).text()).toBe('post.menu.delete')
+        expect(
+          items
+            .at(1)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post-menu-delete')
+        items.at(1).trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
+      })
+
+      it('admin can pin unpinned post', () => {
+        getters['auth/isAdmin'] = () => true
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+            pinnedBy: null,
+          },
+        })
+        let items = wrapper.findAll('.ds-menu-item')
+        expect(items.at(0).text()).toBe('post.menu.pin')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post-menu-pin')
+        items.at(0).trigger('click')
+        expect(wrapper.emitted('pinPost')).toEqual([
+          [
+            {
+              id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+              pinnedBy: null,
+            },
+          ],
+        ])
+      })
+
+      it('admin can unpin pinned post', () => {
+        const wrapper = openContentMenu({
+          isOwner: false,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+            pinnedBy: 'someone',
+          },
+        })
+        let items = wrapper.findAll('.ds-menu-item')
+        // for (var i = 0; i < items.length; i++) { console.log(items.at(i).html()) }
+        expect(items.at(0).text()).toBe('post.menu.unpin')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post-menu-unpin')
+        items.at(0).trigger('click')
+        expect(wrapper.emitted('unpinPost')).toEqual([
+          [
+            {
+              id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+              pinnedBy: 'someone',
+            },
+          ],
+        ])
+      })
+    })
+
+    describe('comment', () => {
+      it('owner can edit and delete comment', () => {
+        const wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'comment',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+        let items = wrapper.findAll('.ds-menu-item')
+        expect(items).toHaveLength(2)
+        expect(items.at(0).text()).toBe('comment.menu.edit')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/comment-menu-edit')
+        items.at(0).trigger('click')
+        expect(wrapper.emitted('showEditCommentMenu')).toEqual([[true]])
+        expect(items.at(1).text()).toBe('comment.menu.delete')
+        expect(
+          items
+            .at(1)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/comment-menu-delete')
+        items.at(1).trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
```


same as :point_up_2:

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2019-11-13T04:37:46Z" title="Wednesday, November 13th 2019, 5:37:46 am +01:00">Nov 13, 2019</time>_

---


I reduced the expectations to only one per it block. Now I am only checking that the methods are called correctly, and skipped all the tests of text, number of items etc.